python/heterocl/devices.py

```diff
@@ -264,7 +264,11 @@ def __getattr__(cls, key):
             raise DeviceError(key + " not supported")
         tool = tool_table[key]
         return cls(key, devs, host, xcel, tool)
-           
+
+class Project():
```

Comment:
We may use the ENV variable to pass the project path to the `execute` function during compilation. But this solution should be fine.